src/Raven.Server/Documents/Queries/Parser/QueryParser.cs

```diff
@@ -1975,8 +1975,26 @@ private List<QueryExpression> ReadMethodArguments()
                     break;
 
                 if (args.Count != 0)
+                {
+                    if (Scanner.TryScan("as"))
+                    {
+                        if (Scanner.Identifier() == false)
+                            ThrowInvalidQueryException("Missing alias for method after 'as'");
+
+                        var lastArg = args[args.Count - 1];
+                        if (lastArg.Type != ExpressionType.Method)
+                            ThrowInvalidQueryException("Alias can only be applied on method");
```

Comment:
Add the alias name here, to make the error better

Reply:
Done